### PR TITLE
can: add ncv7356 SW transceiver driver

### PR DESCRIPTION
**Pull request by vincent-d**
Follow up, based on #5793

This is the NCV7356 (https://www.onsemi.com/pub/Collateral/NCV7356-D.PDF) single wire CAN transceiver driver.

### CODE REVIEW

**Comment by aabadie** [comment]
needs rebase

**Comment by vincent-d** [comment]
Rebased since #6925 has been merged

File: drivers/Makefile.dep [hunk 1]
```diff
@@ -313,3 +313,8 @@ ifneq (,$(filter tja1042,$(USEMODULE)))
   USEMODULE += can_trx
   FEATURES_REQUIRED += periph_gpio
 endif
+
+ifneq (,$(filter nvc7356,$(USEMODULE)))
```
Comment (aabadie) [inline drivers/Makefile.dep]: Please rebase and follow alphabetical order

File: drivers/include/ncv7356.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ * Copyright (C) 2016-2018  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_ncv7356 NCV7356
+ * @ingroup     drivers
```
Comment (aabadie) [inline drivers/include/ncv7356.h]: Not sure, but is there a drivers_can group ?

File: drivers/ncv7356/ncv7356.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+#include "ncv7356.h"
+
+int ncv7356_trx_set_mode(can_trx_t *dev, can_trx_mode_t mode)
+{
+    ncv7356_trx_t *ncv7356 = (ncv7356_trx_t *)dev;
+    int ret;
+
+    switch (mode) {
+    case TRX_NORMAL_MODE:
```
Comment (aabadie) [inline drivers/ncv7356/ncv7356.c]: case should be indented

File: tests/can_trx/Makefile [hunk 4]
```diff
@@ -7,10 +7,14 @@ USEMODULE += ps
...
 
-TRX_TO_BUILD ?= tja1042
+TRX_TO_BUILD ?= tja1042 ncv7356
 
 ifneq (,$(filter tja1042,$(TRX_TO_BUILD)))
     USEMODULE += tja1042
 endif
 
+ifneq (,$(filter ncv7356,$(TRX_TO_BUILD)))
+    USEMODULE += ncv7356
```
Comment (aabadie) [inline tests/can_trx/Makefile]: use 2 spaces indent in makefile

File: drivers/ncv7356/ncv7356.c (hunk 3 above)
Comment (vincent-d) [reply drivers/ncv7356/ncv7356.c]: there are some places where cases are not indented, so I guess we can keep it like that.

**Comment by vincent-d** [comment]
@aabadie I addressed your comments.

**Comment by jcarrano** [comment]
> **un-tested ACK**

Could we not do this again, please?

**Comment by JannesVolkens** [comment]
@vincent-d Where can I find the initialization of mode0_pin and mode1_pin? Or am I missing something?

**Comment by vincent-d** [comment]
@JannesVolkens the pins are initialized in the init function, then the initial value is set with a call to `ncv7356_trx_set_mode()`.

**Comment by PeterKietzmann** [comment]
I guess the question was more about where the pins are configured. Usually, drivers provide a [`params`](https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT#default-device-configuration) file with a default config. This is not the case here.

**Comment by vincent-d** [comment]
oops, sorry for the misunderstanding.

Indeed, one needs to initialize an `ncv7356_trx_t` struct and pass it to the driver.

I have this kind of code in my board config:
```
ncv7356_trx_t trx_sw_can = {
    .trx        = {
        .driver = &ncv7356_driver,
    },
    .mode0_pin  = CAN_MODE0_PIN,
    .mode1_pin  = CAN_MODE1_PIN,
};
```

Then in the `can_params.h` file I have something like this:
```
static const candev_params_t candev_params[] = {
    {
        .name = "can1",
        .trx = (can_trx_t *)&trx_sw_can,
    },
}
```

So the transceiver is used by the CAN stack automatically.

**Comment by PeterKietzmann** [comment]
Thanks for sharing!

**Comment by JannesVolkens** [comment]
I tested the driver for the NCV7356 transceiver using the existing test class can_trx and was able to verify the actual setting and clearing of the two mode pins. I have attached a screenshot from a logic analyzer showing the two lines connected to MODE0 and MODE1, where Channel 0 is MODE0 and Channel 1 is MODE1. You can see the initialization of the device and setting of the sleep mode, high-speed mode, high voltage wake-up mode and normal mode in the given order. The other screenshot shows the terminal and the different printouts of the can_trx test class.

NCV7356TEST

NCV7356TESTING

**Pull request merged**